app/workers/backend_storage_rewrite_worker.rb

```diff
@@ -1,25 +1,20 @@
+# frozen_string_literal: true
+
 class BackendStorageRewriteWorker
   include Sidekiq::Worker
   sidekiq_options queue: :low
 
-  def self.enqueue_all(providers)
-    batch = Sidekiq::Batch.new
-    batch.description = 'Rewriting Backend Storage'
+  ENQUEUER = ->(class_name, ids) { BackendStorageRewriteWorker.perform_async(class_name, ids) }
 
-    providers.select(:id).find_in_batches do |group|
-      batch.jobs do
-        group.each do |provider|
-          perform_async(provider.id)
-        end
-      end
-    end
+  def self.enqueue_all
+    Backend::StorageRewrite::Processor.new(enqueuer: ENQUEUER).rewrite_all
   end
 
   def self.enqueue(provider_id)
-    perform_async(provider_id)
+    Backend::StorageRewrite::Processor.new(enqueuer: ENQUEUER).rewrite_provider(provider_id)
```

Comment:
I think this use case can't happen, there's no rake task for this, and the only call to `BackendStorageRewriteWorker#enqueue` I found in the code is the test you added below. So you could cleanup this method, and the test,

Reply:
it might be useful to have this for situation when we want to do this with a provider as a convenience method from rails console. e.g. in stg

Reply:
but it can be done sync, got what you mean

Reply:
But it might be useful to trigger enqueuing for a single provider through Rails console, for example.